src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs

```diff
@@ -42,7 +42,15 @@ public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>?
                 && bool.TryParse(pair.Value, out bool optionValue)
                 && optionValue) ?? false;
 
-            return Task.FromResult<IDynamicEnumValuesGenerator>(new StartupObjectsEnumGenerator(_workspace, _unconfiguredProject, includeEmptyValue));
+            bool searchForEntryPointsInFormsOnly = false;
+
+            if (_unconfiguredProject.Capabilities.Contains(ProjectCapability.VisualBasic))
+            {
+                searchForEntryPointsInFormsOnly = true;
+                includeEmptyValue = false;
```

Comment:
Super minor: this `if` branch overwrites any value set above, so it is possible to move that work into an `else` branch here and not do that work for VB projects.

Also, why does this vary by language? It'd be nice to call that out in a comment, as the rationale for the behaviour here isn't self-evident.

Reply:
If we're going to check project capabilities and set these values in code, a more idiomatic approach would be to split this type in two: one implementation for C# (with `[AppliesTo(ProjectCapability.CSharp)]`) and one for VB (with `[AppliesTo(ProjectCapability.VisualBasic)]`). They would both be tagged with `[ExportDynamicEnumValuesProvider("StartupObjectsEnumProvider")]`.

I think an even better alternative, though, is to let this be set by an option in the `Rule`, like how we handle `includeEmptyValue`. This gives us more control over the behavior--we can have some default options in the base ApplicationPropertyPage.xaml and then extend/override those through project-type-specific .xaml files.

For example, the behavior with this PR will be to only ever show WinForms entry points in VB projects--even for VB console and WPF apps. We could instead have a "VB WinForms"-specific Application property page that extends the default one and changes the search behavior for only those.

Reply:
> If we're going to check project capabilities and set these values in code, a more idiomatic approach would be to split this type in two: one implementation for C# (with [AppliesTo(ProjectCapability.CSharp)]) and one for VB (with [AppliesTo(ProjectCapability.VisualBasic)]). They would both be tagged with [ExportDynamicEnumValuesProvider("StartupObjectsEnumProvider")].

Totally, these changes were intended to quickly test the EntryPointFinderService 😃

Reply:
> I think an even better alternative, though, is to let this be set by an option in the Rule, like how we handle includeEmptyValue. This gives us more control over the behavior--we can have some default options in the base ApplicationPropertyPage.xaml and then extend/override those through project-type-specific .xaml files.

That's a good point I haven't considered.